core/editor/src/input/mouse.rs

```diff
@@ -13,6 +87,28 @@ impl ViewportPosition {
 		let y_diff = other.y as i64 - self.y as i64;
 		f64::sqrt((x_diff * x_diff + y_diff * y_diff) as f64)
 	}
+	pub fn to_document_position(&self, document_transform: &DocumentTransform, apply_rotation: bool) -> DocumentPosition {
```

Comment:
What's difference between using apply_rotation and not? Intuitively there should only be a single way to transform viewport coordinates to document coordinates. Both cases seem to do a rotation here.